Cleanup debianization (Buster) [REVPI-1688]

iluminat23 opened the pull request.

Commit:
Use debhelper-compat instead of the compat file

The use of the compat file is deprecated and should be avoided. Use
debhelper-compat in the Build-Depends instead.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Avoid newer Standards-Version then provided by the debhelper version

The debhelper of Buster support as newest Standards-Version only 4.3.0.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Reorder the control file contents to match our other packages

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Tell the package build we don't need root or fakeroot

We don't need root or fakeroot to build this package. Add
"Rules-Requires-Root: no" to the control file.

https://www.debian.org/doc/debian-policy/ch-controlfields.html#rules-requires-root

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Use KUNBUS GmbH as maintainer

Use KUNBUS GmbH <support@kunbus.com> as maintainer in the control file.
As we don't have dedicated maintainers for packages. And we use KUNBUS
GmbH for all packages as maintainer.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Use dh_python for handling the installation

The dh_python debhelper should be used to install python applications.
Use ${python3:Depends} instead of depending on a fixed python version.
When using dh_python we also need to build depend on python3.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Remove broken watch file

As we currently don't use the watch files. Just remove the broken watch
file.

https://wiki.debian.org/debian/watch#GitHub
https://lists.debian.org/debian-devel/2022/09/msg00224.html

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Use correct license for upstream source

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Set debian/* under GPL2+ as we usually do

Licensing the debian/* parts as LGPL-3.0+ was an accident. It was always
meant to be licensed under GPL-2+.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Add the upstream-branch to the gbp.conf

As the default upstream-branch is upstream it wouldn't be necessary, but
to be consistent with our other repos, add it to the config. If the
branch changes it is less likely we will forget to add the
upstream-branch to the config.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Remove useless comments from rules file

The rules file contains some comments wich are left over from the
template file. Remove these comments as they have no value.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

Commit:
Drop --with-systemd from rules file

The dh_systemd (and --with systemd) is deprecated. The debhelper now
have native support for systemd and the dh_systemd package is not needed
any more.
So we can drop the --with systemd from the rules file.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

naruxde commented on debian/control:
```diff
@@ -1,23 +1,24 @@
 Homepage: https://github.com/erminas/noderedrevpinodes-server
 Vcs-Browser: https://github.com/RevolutionPi/noderedrevpinodes-server
 Vcs-Git: https://github.com/RevolutionPi/noderedrevpinodes-server.git
+Build-Depends: debhelper-compat (= 12), dh-python, python3
```
```suggestion
Build-Depends: debhelper-compat (= 12), dh-python
```
python3 is a dependency of dh-python. If the upstream project uses setup tools, we have to add python3-all, but they don't even have a setup.py file :D

iluminat23 replied:
Without the build-dep to python3 I get the following lintian error:
```
Now running lintian noderedrevpinodes-server_1.0.4-1+revpi10+1_armhf.changes ...
E: noderedrevpinodes-server source: missing-build-dependency-for-dh-addon python3 => python3:any | python3-all:any | python3-dev:any | python3-all-dev:any | dh-sequence-python3
```

RamiGspo reviewed (commented):
Typo error in Commit [032be9b](https://github.com/RevolutionPi/noderedrevpinodes-server/pull/4/commits/032be9b24929b9aeb5b6403bd80ceb1ff514026a)  alwasy -> always

After that and what Sven commented, it looks good to me.

naruxde commented on debian/control:
```diff
@@ -1,23 +1,24 @@
 Depends:
  ${misc:Depends},
- python3 (>= 3.5),
+ ${python3:Depends},
  python3-revpimodio2 (>= 2.5.1),
  python3-bcrypt (>= 3.1.2),
  python3-cryptography (>= 1.7.1),
  python3-websockets(>= 3.2-2),
- python3-distro(>=1.0.0)
+ python3-distro(>= 1.0.0)
```
Shouldn't we add a reformat commit, because we use at least two spaces as indention in our other packages?

RamiGspo replied:
I had to laugh... well played

Commit:
Fix indention of Depends block

The Depends block is indented with only one space. Indent with two
spaces to make it better readable.

Also add space between comparator and version for python3-distro.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

naruxde commented on debian/changelog:
```diff
@@ -1,3 +1,21 @@
+  * Remove useless comments from rules file
+  * Drop --with-systemd from rules file
+  * Use dh_python for handling the installation
+  * Add missing space (non-functional change)
```
```suggestion
  * Fix indention of Depends block
```

Commit:
Create 1.0.4-1+revpi10+1 release

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>
